### PR TITLE
BButton element

### PR DESCRIPTION
**Pull request by thedevelopnik**
AC:
* Active Buttons can be selected
  * If multiple buttons are available to be selected there should be the ability to highlight the active page/section/button.
* Inactive buttons should be able to be displayed, however, not selectable.
* Buttons should be able to be labeled.
* Buttons should be able to be resized as necessary.
* Button colors should be adjustable.

This accepts `disabled` prop, allows colors to be adjusted via passing in classes, has an aria-label optional prop, and can take height and width props that get translated into a scoped style.

### CODE REVIEW

File: src/components/BButton/BButton.vue [hunk 1]
```diff
@@ -1,18 +1,75 @@
 <template>
-  <button>{{text}}</button>
+  <button
+    v-bind:class="computedClass"
```
Comment (mlear) [inline src/components/BButton/BButton.vue]: i thought we were going with shorthand?

Comment (thedevelopnik) [reply src/components/BButton/BButton.vue]: I apparently missed that in the conversation, will change

Comment (eleanormal) [reply src/components/BButton/BButton.vue]: Oh yeah, I think we said that while you were out of the room

File: src/components/BButton/BButton.vue [hunk 2]
```diff
@@ -1,18 +1,75 @@
 <template>
-  <button>{{text}}</button>
+  <button
+    :class="computedClass"
```
Comment (eleanormal) [inline src/components/BButton/BButton.vue]: I like calling out that it's computed here!

File: src/components/BButton/BButton.vue [hunk 3]
```diff
@@ -1,18 +1,75 @@
...
+    :disabled="disabled"
+    :type="type"
+    :height="height"
```
Comment (eleanormal) [inline src/components/BButton/BButton.vue]: Height and weight are handled by style, are they not?

File: src/components/BButton/BButton.vue [hunk 4]
```diff
@@ -1,18 +1,75 @@
 <template>
-  <button>{{text}}</button>
+  <button
+    :class="computedClass"
+    :disabled="disabled"
```
Comment (eleanormal) [inline src/components/BButton/BButton.vue]: Could also do `v-bind="{disabled, type, aria-label}"`. Don't know if we have a preference there yet

File: src/components/BButton/BButton.vue (hunk 3 above)
Comment (thedevelopnik) [reply src/components/BButton/BButton.vue]: yep, fixed that, thanks

File: src/components/BButton/BButton.vue (hunk 4 above)
Comment (thedevelopnik) [reply src/components/BButton/BButton.vue]: tried that and I was getting an error

**Comment by thedevelopnik** [comment]
wrapper.classes() is acting weird, returning this huge array of Symbols and Functions, rather than just the classes. Skipped out that test for now to get more eyes on it.

File: src/components/BButton/BButton.test.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    })
+
+    expect(wrapper.hasStyle('height', '50px')).toBe(true)
+    expect(wrapper.hasStyle('width', '100px')).toBe(true)
+  })
+
+  it('is enabled when disabled prop is not passed', () => {
+    const wrapper = shallow(BButton, {})
+
+    expect(wrapper.props().disabled).toBe(false)
```
Comment (eleanormal) [inline src/components/BButton/BButton.test.js]: Isn't this just testing that disabled didn't get passed in? It doesn't seem to be testing that the button is actually using the disabled value

File: src/components/BButton/BButton.vue (hunk 4 above)
Comment (eleanormal) [reply src/components/BButton/BButton.vue]: Huh, that's surprising

File: src/components/BButton/BButton.test.js (hunk 5 above)
Comment (thedevelopnik) [reply src/components/BButton/BButton.test.js]: yeah I thought about that as I was writing it. Need to think more about how to test if it's actually disabled and vice versa. Maybe I can use the click function somehow.

Comment (stevenmwade) [reply src/components/BButton/BButton.test.js]: Testing if a button is disabled if it has `disabled` seems like too much. That sounds like a) we don't know how to write a Vue component or b) we are now testing actual HTML. If you correctly bind `disabled` the element should have it and should behave according to the HTML specs. In my opinion, a snapshot test works fine.

File: src/components/BButton/BButton.test.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  })
+
+  it('has an aria-label if one is passed', () => {
+    const wrapper = shallow(BButton, {
+      propsData: {
+        'aria-label': 'submit button'
+      }
+    })
+
+    expect(wrapper.props()['aria-label']).toBe('submit button')
```
Comment (eleanormal) [inline src/components/BButton/BButton.test.js]: This should also be tested with a snapshot.

File: src/components/BButton/BButton.vue [hunk 7]
```diff
@@ -1,18 +1,71 @@
...
+      type: String,
+      required: false
+    }
+  },
   data () {
     return {}
+  },
+  computed: {
+    computedClass () {
+      // if any classes are passed through, provided them to the component as an array of strings
```
Comment (eleanormal) [inline src/components/BButton/BButton.vue]: This comment no longer seems true

**Pull request merged**